src/common/com/intellij/plugins/haxe/ide/HaxeLookupElement.java

```diff
@@ -107,6 +112,28 @@ public void renderElement(LookupElementPresentation presentation) {
     }
   }
 
+  @Override
+  public void handleInsert(InsertionContext context) {
+    HaxeMemberModel memberModel = HaxeMemberModel.fromPsi(myComponentName);
+    boolean hasParams = false;
+    boolean isMethod = false;
+    if (memberModel != null) {
+      if (memberModel instanceof HaxeMethodModel) {
+        isMethod = true;
+        HaxeMethodModel methodModel = (HaxeMethodModel)memberModel;
+        hasParams = !methodModel.getParametersWithContext(this.context).isEmpty();
+      }
+    }
+
+    if (isMethod) {
+      final LookupElement[] allItems = context.getElements();
+      final boolean overloadsMatter = allItems.length == 1 && getUserData(FORCE_SHOW_SIGNATURE_ATTR) == null;
+      JavaCompletionUtil.insertParentheses(context, this, overloadsMatter, hasParams);
+    }
+  }
+  // Workaround for IDEA 14. There must be JavaCompletionUtil.FORCE_SHOW_SIGNATURE_ATTR
+  private static final Key<Boolean> FORCE_SHOW_SIGNATURE_ATTR = Key.create("forceShowSignature");
```

Comment:
Instead, you could do this:
```
private static final Key<Boolean> FORCE_SHOW_SIGNATURE_ATTR =
    IdeaTarget.IS_VERSION_15_COMPATIBLE ? JavaCompletionUtil.FORCE_SHOW_SIGNATURE_ATTR 
                                        : Key.create("forceShowSignature");
```
If you do so, then we're future-proofed against any actual change to the key.  V14 won't ever have anything stored using that key, so it will always be null.